### PR TITLE
Fix: gitee code analysis shadow err

### LINKED ISSUE #514: [Imp] modify codes by code review of gitee
**What would you like to be added**:

improve the code quality scanned by gitee, from  the following two links,

[gitee_scans](https://gitee.com/dubbogo/dubbo-go/gitee_scans/report?task_id=723471)
[quality_analyses](https://gitee.com/dubbogo/dubbo-go/quality_analyses?platform=sonar_qube)

**Why is this needed**:

Improve the code quality.

### PR DESCRIPTION
**Pull request by georgehao**
**What this PR does**:
fix [gitee_scans](https://gitee.com/dubbogo/dubbo-go/gitee_scans/report?task_id=723471) shadow error

**Which issue(s) this PR fixes**:
Fixes #514

### CODE REVIEW

File: cluster/directory/base_directory.go [hunk 1]
```diff
@@ -92,7 +92,7 @@ func (dir *BaseDirectory) SetRouters(urls []*common.URL) {
 			factory := extension.GetRouterFactory(url.Protocol)
 			r, err := factory.NewRouter(url)
 			if err != nil {
-				logger.Errorf("Create router fail. router key: %s, error: %v", routerKey, url.Service(), err)
+				logger.Errorf("Create router fail. router key: %s, url:%s, error: %s", routerKey, url.Service(), err.Error())
```
Comment (zouyx) [inline cluster/directory/base_directory.go:95]:
```suggestion
				logger.Errorf("Create router fail. router key: %s, url:%s, error: %+v", routerKey, url.Service(), err.Error())
```

File: cluster/directory/static_directory.go [hunk 2]
```diff
@@ -61,7 +61,7 @@ func (dir *staticDirectory) IsAvailable() bool {
 // List List invokers
 func (dir *staticDirectory) List(invocation protocol.Invocation) []protocol.Invoker {
 	l := len(dir.invokers)
-	invokers := make([]protocol.Invoker, l, l)
+	invokers := make([]protocol.Invoker, l)
```
Comment (zouyx) [inline cluster/directory/static_directory.go:64]: why delete capacity ?

File: filter/filter_impl/hystrix_filter.go [hunk 3]
```diff
@@ -124,7 +124,7 @@ func (hf *HystrixFilter) Invoke(ctx context.Context, invoker protocol.Invoker, i
 	_, _, err := hystrix.GetCircuit(cmdName)
 	configLoadMutex.RUnlock()
 	if err != nil {
-		logger.Errorf("[Hystrix Filter]Errors occurred getting circuit for %s , will invoke without hystrix, error is: ", cmdName, err)
+		logger.Errorf("[Hystrix Filter]Errors occurred getting circuit for %s , will invoke without hystrix, error is: %s", cmdName, err.Error())
```
Comment (zouyx) [inline filter/filter_impl/hystrix_filter.go:127]:
```suggestion
		logger.Errorf("[Hystrix Filter]Errors occurred getting circuit for %s , will invoke without hystrix, error is: %+v", cmdName, err.Error())
```

File: registry/kubernetes/registry_test.go [hunk 4]
```diff
@@ -68,7 +68,7 @@ func (s *KubernetesRegistryTestSuite) TestSubscribe() {
 	time.Sleep(1e9)
 
 	go func() {
-		err := r.Register(url)
+		err = r.Register(url)
```
Comment (zouyx) [inline registry/kubernetes/registry_test.go:71]: i think assign to `err` outside goroutine should cause concurrent problems

File: cluster/directory/static_directory.go (hunk 2 above)
Comment (georgehao) [reply cluster/directory/static_directory.go:64]: gosamle detect it. I also think the `capactity` no useful
```
static_directory.go:64:39: S1019: should use make([]protocol.Invoker, l) instead (gosimple)
invokers := make([]protocol.Invoker, l, l)
	                                     ^
```

**Pull request merged**